### DISCUSSION TITLE
VueSanize from vue-sanitize

### DISCUSSION THREAD

**Discussion by brogits** [quasarframework/quasar#16968] [General - Components / Directives / etc] [answered]
Hello,

I'm new with quasar, but I could tell I will be hooked on this framework.

Anyhow had anyone had success implementing an html sanitizer? I'm trying to use v-html on a component and I know its risky but I'm willing to go that way as long as I have some code security I can implement like sanitizer.

Any idea or suggestion which way I should go with this task?

Thanks and more power
- brogits here

**Comment by yusufkandemir** [comment, marked as answer]
Hello, you can use https://github.com/cure53/DOMPurify, here are two examples:

1. `v-html`
	```html
	<template>
	  <div v-html="purified" />
	</template>
	
	<script setup>
	import * as DOMPurify from 'dompurify';
	import { computed, ref } from 'vue';

	// imagine this changes somehow, maybe coming from an API
	const reactiveHTML = ref("<img src='#' onerror=alert(1)>");
	const purified = computed(() => DOMPurify.sanitize(reactiveHTML.value));
	</script>
	```
2. `Notify` plugin
	```html
	<script setup>
	import * as DOMPurify from 'dompurify';
	import { Notify } from 'quasar';
	
	Notify.create({
	  message: DOMPurify.sanitize("<img src='#' onerror=alert(1)>"),
	  html: true,
	})
	</script>
	```

You can also use `require-trusted-types-for` CSP directive along with it, see
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/require-trusted-types-for#examples

**Reply by brogits** [reply]
Cool thanks a lot
